cli/cook/querying.py

```diff
@@ -190,19 +227,91 @@ def __get_latest_instance(job):
     raise Exception(f'Job {job["uuid"]} currently has no instances.')
 
 
-def query_unique_and_run(clusters, job_or_instance_uuid, command_fn):
+def query_unique_and_run(clusters, entity_ref, command_fn):
     """Calls query_unique and then calls the given command_fn on the resulting job instance"""
-    query_result = query_unique(clusters, job_or_instance_uuid)
-    if query_result['type'] == 'job':
+    query_result = query_unique(clusters, entity_ref)
+    if query_result['type'] == Types.JOB:
         job = query_result['data']
         instance = __get_latest_instance(job)
         directory = mesos.retrieve_instance_sandbox_directory(instance, job)
         command_fn(instance, directory)
-    elif query_result['type'] == 'instance':
+    elif query_result['type'] == Types.INSTANCE:
         instance, job = query_result['data']
         directory = mesos.retrieve_instance_sandbox_directory(instance, job)
         command_fn(instance, directory)
     else:
         # This should not happen, because query_unique should
         # only return a map with type "job" or type "instance"
-        raise Exception(f'Encountered error when querying for {job_or_instance_uuid}.')
+        raise Exception(f'Encountered error when querying for {entity_ref}.')
+
+
+def parse_entity_refs(clusters, ref_strings):
+    """
+    Given a collection of entity ref strings, returns a list of entity ref maps, where each map has
+    the following shape:
+
+      {'cluster': ..., 'type': ..., 'uuid': ...}
+
+    The cluster field is the name of a cluster, the type field can be either job, instance, or group,
+    and the uuid field is the uuid of the entity in question.
+
+    An entity ref string can either be simply a UUID, in which case all configured clusters and all
+    three types will be queried, or it can be a Cook Scheduler URL that represents the way to retrieve
+    that specific entity.
+
+    Some examples of valid entity ref strings follow:
+
+      5ab383b1-5b8b-4483-b2b2-45126923f4df
+      http://cluster2.example.com/jobs/5ab383b1-5b8b-4483-b2b2-45126923f4df
+      http://cluster2.example.com/jobs?uuid=5ab383b1-5b8b-4483-b2b2-45126923f4df
+      http://cluster3.example.com/instances/5ab383b1-5b8b-4483-b2b2-45126923f4df
+      http://cluster3.example.com/instances?uuid=5ab383b1-5b8b-4483-b2b2-45126923f4df
+      http://cluster4.example.com/groups/5ab383b1-5b8b-4483-b2b2-45126923f4df
+      http://cluster4.example.com/groups?uuid=5ab383b1-5b8b-4483-b2b2-45126923f4df
```

Comment:
Why do we want to support both a `/«uuid»` and `?uuid=«uuid»`? Since all of these endpoints are new anyway, why not just do away with the non-REST-like use of query parameters to identify the resource? If we want to be able to identify more than one group in the same URL, maybe we should have a special identifier in place of the UUID, e.g.: `http://cluster2.example.com/jobs/multiple?uuid=...&uuid=...&uuid=...`